icubam/www/handlers/upload_csv.py

```diff
@@ -1,48 +0,0 @@
-import os
-import re
-import tornado.web
-import logging
-import time
-
-from icubam.db import store
-from icubam.www.handlers import base
-
-
-def clean_path(f_name):
-  # Sanitize the name some:
-  f_name = f_name.lower()
-  f_name = os.path.normpath(f_name)
-  f_name = os.path.split(f_name)[1]
-  f_name, f_ext = os.path.splitext(f_name)
-  # Clean out non-alphanumerics (need to split ext first or we lose the '.'):
-  f_name, f_ext = [re.sub(r'\W+', '', n) for n in [f_name, f_ext]]
-  if f_ext == '':
-    return f_name
-  else:
-    return f"{f_name}.{f_ext}"
-
-
-class UploadHandler(base.APIKeyProtectedHandler):
-  """Accept file uploads over POST."""
-
-  ROUTE = "/upload_csv"
-  API_COOKIE = 'api'
-  ACCESS = [store.AccessTypes.UPLOAD, store.AccessTypes.ALL]
-
-  def initialize(self, upload_path, config, db_factory):
-    self.upload_path = upload_path
-    super().initialize(config, db_factory)
-
-  @tornado.web.authenticated
-  def post(self):
-    file = self.request.files["file"][0]
-    file_name = clean_path(file["filename"])
```

Comment:
I like that you no longer take the filename from the user, it is often dangerous from a security standpoint (even if it was sanitized here, you never know how these functions work when people play with null characters and the likes).

Reply:
yeah, I'm also glad we got rid of that.  I'm still a bit worried about pandas reading the provided CSV as-is, esp. because you can't force pandas to consider it as a file and not a URL (or I'm not sure how), so someone could potentially feed in a huuuuge URL and overload pandas' CSV reading.  I was thinking of going with csvlib here for that reason, @tgy or @rth any thoughts?  The user /is/ authenticated so there is also a certain level of trust.